### ISSUE TITLE
LAAF questions

### ISSUE THREAD

**Issue by EvelyneCalista** [paulhager/MMCL-Tabular-Imaging#16]
Hi Hager,

Thanks for your great work.
I would like to ask for training the Laaf:
1. do the corruption augmentation and one hot encoder also applied for the label?
2. Do the loss function for Laaf is using "supcon_loss_custom.py" this py file? I am getting confused with the other two py files "supcon_loss_clip_binary.py" and "supcon_loss_clip.py"

Thank you.

**Comment by paulhager** [comment]
Hi Evelyne,

1. Yes, the label is treated as a normal feature that can be corrupted. I did not experiment with leaving it out of the corruption strategy, this might be worth taking a look at!
2. You can use either the supcon loss (yes, its supcon_loss_custom, sorry for the confusion) or a normal NTXent loss, both work. In the paper we showed that the effect is often additive, so supcon loss + LaaF should return best results.

**Comment by anikmuhury** [comment]
Hello Hager,

Thank you for this awesome research work. 
I have trouble configure the environment properly from the yaml file as some of the libraries are deprecated. Is it possible for you to share your environment as setup.exe file, so that I can mirror it.
I appreciate your help.
Thank you

**Comment by Coello-dev** [comment]
Hello Hager,

it's still unclear to me how LaaF was implemented in the code. From your previous message ("Yes, the label is treated as a normal feature that can be corrupted.") it seems the label and feature concatenation is done in the dataloader.
Nevertheless, after checking the file "ContrastiveImagingAndTabularDataset.py" I'm not able to find where this happens.

Additionally, I have checking the notebooks which generate the data and I'm not able to see the label getting added at any point to the csv files.

Could you explicitly point out where the concatenation is done?

Thank you
